sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ResourceTokenTest.java

```diff
@@ -397,6 +411,40 @@ public void readDocumentFromCollPermissionWithDiffPartitionKey_WithException() t
         }
     }
 
+    @Test(groups = { "simple" }, dataProvider = "queryItemPermissionData", timeOut = TIMEOUT)
+    public void queryItemFromResourceToken(DocumentCollection documentCollection, Permission permission, PartitionKey partitionKey) throws Exception {
+
+        AsyncDocumentClient asyncClientResourceToken = null;
+        try {
+            List<Permission> permissionFeed = new ArrayList<>();
+            permissionFeed.add(permission);
+
+            asyncClientResourceToken = new AsyncDocumentClient.Builder()
+                .withServiceEndpoint(TestConfigurations.HOST)
+                .withConnectionPolicy(ConnectionPolicy.getDefaultPolicy())
+                .withConsistencyLevel(ConsistencyLevel.SESSION)
+                .withMasterKeyOrResourceToken(permission.getToken())
+                .build();
+
+            CosmosQueryRequestOptions queryRequestOptions = new CosmosQueryRequestOptions();
+            queryRequestOptions.setPartitionKey(partitionKey);
+            Flux<FeedResponse<Document>> queryObservable =
```

Comment:
Can we add a test to use V4 API instead of the context client? That would also catch any bugs if we have in the upper layers.

Reply:
Yea, we do have tests for V4: ResourceTokenTests

Reply:
Refactored from ResourceTokenTests to ResourceTokenTestForV4